frontend/src/components/upload/modals/showCompletedUploadModal.tsx

```diff
@@ -27,22 +28,34 @@ const Body = ({ share, appUrl }: { share: Share; appUrl: string }) => {
   const modals = useModals();
   const router = useRouter();
 
+  const [checkState, setCheckState] = useState(false);
+  const timerRef = useRef<number | ReturnType<typeof setTimeout> | undefined>(
+    undefined
+  );
+
   const link = `${appUrl}/share/${share.id}`;
+
+  const copyLink = () => {
+    clipboard.copy(link);
+    toast.success("Your link was copied to the keyboard.");
+    if (timerRef.current) clearTimeout(timerRef.current);
+    timerRef.current = setTimeout(() => {
+      setCheckState(false);
+    }, 1500);
+    setCheckState(true);
+  };
+
   return (
     <Stack align="stretch">
       <TextInput
         readOnly
         variant="filled"
         value={link}
+        onClick={copyLink}
         rightSection={
           window.isSecureContext && (
-            <ActionIcon
-              onClick={() => {
-                clipboard.copy(link);
-                toast.success("Your link was copied to the keyboard.");
-              }}
-            >
-              <TbCopy />
+            <ActionIcon onClick={copyLink}>
+              {checkState ? <TbChecks /> : <TbCopy />}
```

Comment:
```suggestion
              {checkState ? <TbCheck /> : <TbCopy />}
```
I think the icon with the single check mark would be a bit cleaner.

Reply:
Ok, I'll create a new component and add it in the other file asap